libs/core/src/lib/avatar-group/avatar-group.component.ts

```diff
@@ -0,0 +1,182 @@
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    Component,
+    ContentChildren,
+    ElementRef,
+    forwardRef,
+    HostBinding,
+    Input,
+    OnChanges,
+    OnDestroy,
+    OnInit,
+    QueryList,
+    ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+
+import { fromEvent, of, Subscription } from 'rxjs';
+import { debounceTime, delay, distinctUntilChanged } from 'rxjs/operators';
+
+import { ColorAccent, Size } from '../utils/public_api';
+import { AvatarGroupItemDirective } from './directives/avatar-group-item.directive';
+
+export type AvatarGroupType = 'group' | 'individual';
+export type AvatarGroupOverflowButtonColor = 'neutral' | 'random' | ColorAccent;
+
+let avatarGroupUniqueId = 0;
+
+@Component({
+    selector: 'fd-avatar-group',
+    templateUrl: './avatar-group.component.html',
+    styleUrls: ['./avatar-group.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.Default
+})
+export class AvatarGroupComponent implements OnChanges, OnInit, AfterViewInit, OnDestroy {
+    /** Id of the Avatar Group. */
+    @Input()
+    @HostBinding('attr.id')
+    id = `fd-avatar-group-${avatarGroupUniqueId++}`;
+
+    /** The size of the Avatar Group. Options include: *xs*, *s*, *m*, *l* and *xl* (default: *s*). */
+    @Input()
+    size: Size = 's';
+
+    /** The type of the Avatar Group. Options include: *group* and *individual* (default: *group*). */
+    @Input()
+    type: AvatarGroupType = 'group';
+
+    /** Aria-label for Avatar Group. */
+    @Input()
+    ariaLabel: string = null;
+
+    /** Counter for all avatars. */
+    allItemsCount = 0;
+
+    /** Counter for visible in main content avatars. */
+    visibleItemsCount = 0;
```

Comment:
I'm not sure this variable serves much of a purpose.  And if there ever is a case where the application developer needs this info can't they just `allItemsCount - overflowItemsCount`